scrapy/extensions/httpcache.py

```diff
@@ -326,8 +326,12 @@ def _get_request_path(self, spider, request):
     def _read_meta(self, spider, request):
         rpath = self._get_request_path(spider, request)
         metapath = os.path.join(rpath, 'pickled_meta')
-        if not os.path.exists(metapath):
-            return  # not found
+        try:
+            os.stat(metapath)
+        except OSError as exc:
+            if exc.errno == os.errno.ENOENT:
+                return  # not found
+            raise
         mtime = os.stat(metapath).st_mtime
```

Comment:
could this line be used in the `try:` section above to save an `os.stat()` call?